uberenv.py

```diff
@@ -262,6 +255,17 @@ def parse_args():
                       default=False,
                       help="Only install (using pre-setup Spack).")
 
+    # Spack Environment directory
+    parser.add_option("--spack-env-name",
+                      dest="spack_env_name",
+                      default="spack_env",
+                      help="The created Spack Environment directory. Will create directory automatically if it doesn't exist.")
```

Comment:
```suggestion
    # Spack Environment name
    parser.add_option("--spack-env-name",
                      dest="spack_env_name",
                      default="spack_env",
                      help="The name of the created Spack Environment.")
```

Reply:
I think it would be helpful to not default to `spack_env` but to actually `spack_env + <time date stamp>`. That way if you run uberenv times, you will not override the same environment file. This would be helpful for debugging after several runs.

Reply:
That would break repeated calls to uberenv however (multiple calls iterating towards success, like if a download fails)?

Reply:
That's a distinct possibility. Lets just leave it out of this PR and maybe try it afterwards as a targeted PR. I am not sure, without testing personally, what would happen if you just called uberenv multiple times on the same spec.

Reply:
To be clear - repeated calls to the same build is something I do often with the current uberenv during development.

Reply:
I do that too. 

How about an extra option to append the date/time which is off by default, but could be added to the uberenv-config file?

Reply:
I am fine with just leaving it a generic name like it is now. It was just a random thought that clearly everyone hates. <cries>

Reply:
did suggestion and updated docs as well